posthog/temporal/data_imports/pipelines/postgres/postgres.py

```diff
@@ -173,9 +173,35 @@ def _get_primary_keys(cursor: psycopg.Cursor, schema: str, table_name: str) -> l
     return None
 
 
-def _get_table_chunk_size(
-    cursor: psycopg.Cursor, inner_query: sql.Composed, schema: str, table_name: str, logger: FilteringBoundLogger
-) -> int:
+def _has_duplicate_primary_keys(
+    cursor: psycopg.Cursor, schema: str, table_name: str, primary_keys: list[str] | None
+) -> bool:
+    if not primary_keys or len(primary_keys) == 0:
+        return False
+
+    try:
+        sql_query = cast(
+            LiteralString,
+            f"""
+            SELECT {", ".join(["{}" for _ in primary_keys])}
+            FROM {{}}.{{}}
+            GROUP BY {", ".join([str(i + 1) for i, _ in enumerate(primary_keys)])}
+            HAVING COUNT(*) > 1
+        """,
```

Comment:
We could probably have a `limit 1` here as we only really care if `len(rows) > 0`